src/coreclr/vm/interpexec.cpp

```diff
@@ -155,35 +191,50 @@ CallStubHeader *CreateNativeToInterpreterCallStub(InterpMethod* pInterpMethod)
 
     return pHeader;
 }
-
 #endif // !TARGET_WASM
 
-// Use the NOINLINE to ensure that the InlinedCallFrame in this method is a lower stack address than any InterpMethodContextFrame values.
-NOINLINE void InvokePInvokeMethod(MethodDesc *targetMethod, int8_t *stack, InterpMethodContextFrame *pFrame, int32_t callArgsOffset, int32_t returnOffset, PCODE callTarget)
+#ifdef _DEBUG
+void DBG_PrintInterpreterStack()
 {
-    InlinedCallFrame inlinedCallFrame;
-    inlinedCallFrame.m_pCallerReturnAddress = (TADDR)pFrame->ip;
-    inlinedCallFrame.m_pCallSiteSP = pFrame;
-    inlinedCallFrame.m_pCalleeSavedFP = (TADDR)stack;
-    inlinedCallFrame.m_pThread = GetThread();
-    inlinedCallFrame.m_Datum = NULL;
-    inlinedCallFrame.Push();
+    Thread* pThread = GetThread();
+
+    int32_t frameCount = 0;
 
+    // Get the "capital F" frame and start walking.
+    for (Frame* pFrame = pThread->GetFrame(); pFrame != FRAME_TOP; pFrame = pFrame->PtrNextFrame())
```

Comment:
Is it possible we enter DBG_PrintInterpreterStack with no frame(s) on the thread? I don't know the answer to this, just wondering since you put a call to this in the assert handler iirc

Reply:
Also is it possible we have no pThread yet if we assert on a native thread that hasn't had a managed thread object set up yet, and then ->GetFrame crashes?

Reply:
Right, the guard for no Thread is the issue. No capital F frames is already handled by `pFrame != FRAME_TOP`.